core/tests/performance_framework/perf_services.py

```diff
@@ -221,10 +256,104 @@ def _setup_driver(self, proxy=None, use_proxy=False):
 
             driver = webdriver.Firefox(firefox_profile=firefox_profile)
 
+        self._add_cookie(driver)
+
         return driver
 
+    def _add_cookie(self, driver):
+        if self.dev_appserver_login_cookie:
+            driver.get('%s%s' % (self.BASE_URL, feconf.ROBOTS_TXT_URL))
+            driver.add_cookie(self.dev_appserver_login_cookie)
+
     def _stop_proxy_server(self, server):
         server.stop()
 
     def _stop_driver(self, driver):
         driver.quit()
+
+    def _check_login(self, driver):
```

Comment:
Then I'd suggest adding a docstring explaining what it does, since there's no example of usage.